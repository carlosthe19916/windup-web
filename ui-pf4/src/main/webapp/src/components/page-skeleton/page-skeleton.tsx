import * as React from "react";
import { PageSection, PageSectionVariants } from "@patternfly/react-core";

import { Typography } from "@material-ui/core";
import Skeleton from "@material-ui/lab/Skeleton";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardContent from "@material-ui/core/CardContent";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  card: {
    maxWidth: "100%",
    margin: theme.spacing(0),
  },
  media: {
    height: 300,
  },
}));

export const PageSkeleton: React.FC = () => {
  const classes = useStyles();

  return (
    <React.Fragment>
      <PageSection variant={PageSectionVariants.light}>
        <Typography component="div" variant="body1">
          <Skeleton />
        </Typography>
      </PageSection>
      <PageSection>
        <Card className={classes.card}>
          <CardHeader
            avatar={
              <Skeleton
                animation="wave"
                variant="circle"
                width={60}
                height={60}
              />
            }
            title={
              <Skeleton
                animation="wave"
                height={20}
                width="80%"
                style={{ marginBottom: 6 }}
              />
            }
            subheader={<Skeleton animation="wave" height={20} width="40%" />}
          />
          <Skeleton animation="wave" variant="rect" className={classes.media} />
          <CardContent>
            <React.Fragment>
              <Skeleton
                animation="wave"
                height={20}
                style={{ marginBottom: 6 }}
              />
              <Skeleton animation="wave" height={20} width="80%" />
            </React.Fragment>
          </CardContent>
        </Card>
      </PageSection>
    </React.Fragment>
  );
};
