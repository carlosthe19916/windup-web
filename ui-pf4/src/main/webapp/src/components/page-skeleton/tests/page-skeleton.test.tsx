import * as React from "react";
import { shallow } from "enzyme";
import { PageSkeleton } from "../page-skeleton";

describe("SimplePagination", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<PageSkeleton />);
    expect(wrapper).toMatchSnapshot();
  });
});
