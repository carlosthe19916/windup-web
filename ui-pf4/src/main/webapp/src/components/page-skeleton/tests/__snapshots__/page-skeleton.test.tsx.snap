// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePagination Renders without crashing 1`] = `
<Fragment>
  <PageSection
    variant="light"
  >
    <WithStyles(ForwardRef(Typography))
      component="div"
      variant="body1"
    >
      <WithStyles(ForwardRef(Skeleton)) />
    </WithStyles(ForwardRef(Typography))>
  </PageSection>
  <PageSection>
    <WithStyles(ForwardRef(Card))
      className="makeStyles-card-1"
    >
      <WithStyles(ForwardRef(CardHeader))
        avatar={
          <ForwardRef(WithStyles)
            animation="wave"
            height={60}
            variant="circle"
            width={60}
          />
        }
        subheader={
          <ForwardRef(WithStyles)
            animation="wave"
            height={20}
            width="40%"
          />
        }
        title={
          <ForwardRef(WithStyles)
            animation="wave"
            height={20}
            style={
              Object {
                "marginBottom": 6,
              }
            }
            width="80%"
          />
        }
      />
      <WithStyles(ForwardRef(Skeleton))
        animation="wave"
        className="makeStyles-media-2"
        variant="rect"
      />
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={20}
          style={
            Object {
              "marginBottom": 6,
            }
          }
        />
        <WithStyles(ForwardRef(Skeleton))
          animation="wave"
          height={20}
          width="80%"
        />
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(Card))>
  </PageSection>
</Fragment>
`;
